### PR TITLE
Improve 3D plotting

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
* Fixed aspect ratio
* Less lost white space

Overrides #1652

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
@jeromedockes, if you have time to look at this PR it would be great. It's a continuation of your #1652 PR.

**Comment by GaelVaroquaux** [comment]
It renders well on circle:
* https://2946-1235740-gh.circle-artifacts.com/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/01_plotting/plot_3d_map_to_surface_projection.html#sphx-glr-auto-examples-01-plotting-plot-3d-map-to-surface-projection-py
* https://2946-1235740-gh.circle-artifacts.com/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/01_plotting/plot_surf_atlas.html#sphx-glr-auto-examples-01-plotting-plot-surf-atlas-py
* https://2946-1235740-gh.circle-artifacts.com/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/01_plotting/plot_surf_stat_map.html#sphx-glr-auto-examples-01-plotting-plot-surf-stat-map-py

**Comment by KamalakerDadi** [comment]
Nice improvement. LGTM

File: nilearn/plotting/surf_plotting.py [hunk 1]
```diff
@@ -172,13 +172,14 @@ def plot_surf(surf_mesh, surf_map=None, bg_map=None,
     if axes is None:
         if figure is None:
             figure = plt.figure()
-        axes = figure.add_subplot(111, projection='3d',
-                                  xlim=limits, ylim=limits)
+        axes = Axes3D(figure, rect=[0, 0, 1, 1],
```
Comment (KamalakerDadi) [inline nilearn/plotting/surf_plotting.py]: Nitpick: I would test passing a figure to ```plot_surf```.

**Comment by KamalakerDadi** [comment]
Please add this in whats_new.

**Comment by GaelVaroquaux** [comment]
> Nitpick: I would test passing a figure to plot_surf.

It's tested, on line 163 of test_surf_plotting.py

I've updated the changelog. This should be good to go.

**Comment by bthirion** [comment]
LGTM. B

**Comment by jeromedockes** [comment]
looks great, thanks! there is still a lot of whitespace when saving the figure, but I think that's for another PR.

**Comment by jeromedockes** [comment]
> This should be good to go.

I think so too

**Comment by GaelVaroquaux** [comment]
Thanks for the reviews.

Indeed, there is still room for improvement.

**Comment by KamalakerDadi** [comment]
>I've updated the changelog.

@GaelVaroquaux I cannot see where this contribution is added. Should I add it ?

**Comment by GaelVaroquaux** [comment]
>     I've updated the changelog.

> @GaelVaroquaux I cannot see where this contribution is added. Should I add it ?

I had forgotten to push :/. I've cherry-picked the commit and pushed it
into master.

**Pull request merged**